link-grammar/count.c

```diff
@@ -318,10 +314,6 @@ static Count_bin do_count(fast_matcher_t *mchxt,
 
 				/* Now, we determine if (based on table only) we can see that
 				   the current range is not parsable. */
-				Lmatch = (le != NULL) && (d->left != NULL) &&
-				         do_match(le, d->left, lw, w);
-				Rmatch = (d->right != NULL) && (re != NULL) &&
-				         do_match(d->right, re, w, rw);
 
```

Comment:
Note that the reported speedup was measured on batch runs, and in batch mode `lnull_cnt` is supposed to always be **0**, so the loop is executed only once, and pulling it out of the loop cannot contribute to the speedup.
However, when I tested this pulling out on "detailed batch run" (producing all the linkages, as if it was interactive run) I then observed a speed up which I attributed to this statement move (as sentences which don't get parsed with null count 0 are tried again with lnull_cnt = sent_length).